src/transformers/trainer.py

```diff
@@ -1875,17 +1877,37 @@ def train(
 
         # Load potential model checkpoint
         if isinstance(resume_from_checkpoint, bool) and resume_from_checkpoint:
-            resume_from_checkpoint = get_last_checkpoint(args.output_dir)
-            if resume_from_checkpoint is None:
+            checkpoints = get_checkpoints(args.output_dir, reverse=True)
+            if not checkpoints:
                 raise ValueError(f"No valid checkpoint found in output directory ({args.output_dir})")
+        else:
+            checkpoints = [resume_from_checkpoint] if resume_from_checkpoint else []
 
-        if resume_from_checkpoint is not None:
+        if len(checkpoints) > 0:
             if not is_sagemaker_mp_enabled() and not self.is_deepspeed_enabled and not self.is_fsdp_enabled:
-                self._load_from_checkpoint(resume_from_checkpoint)
+                for checkpoint in checkpoints:
+                    try:
+                        self._load_from_checkpoint(checkpoint)
+                        checkpoints = [checkpoint]
+                        break
+                    except (ValueError, safetensors.SafetensorError) as e:
+                        logger.warning(f"Error loading checkpoint {checkpoint}: {e}\nMoving onto next checkpoint.")
```

Comment:
As much as I may not like it, we may be best just having an `except Exception` here simply because we can't account for everything. 

Similarly down at the `OSError` level